common/content/world_content_service.cpp

```diff
@@ -139,6 +139,51 @@ bool WorldContentService::IsContentFlagEnabled(const std::string &content_flag)
 	return false;
 }
 
+/**
+ * @param content_flag
+ * @return
+ */
+bool WorldContentService::IsContentFlagDisabled(const std::string &content_flag)
+{
+	for (auto &f: GetContentFlags()) {
+		if (f.flag_name == content_flag && f.enabled == false) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+bool WorldContentService::DoEnabledFlagsPass(const char *content_flags) {
```

Comment:
Also pass `std::string` into this

Name should be a bit more consistent with the rest of the class but just imply "many"

```cpp
IsAnyContentFlagEnabled(std::string content_flags)
IsAnyContentFlagDisabled(std::string content_flags)
```

Also; it would be better to split the string into a vector before entering this function and just pass a vector of strings into the function which makes your code easier to work with in the function itself

```cpp
IsAnyContentFlagEnabled(std::vector<std::string> content_flags)
IsAnyContentFlagDisabled(std::vector<std::string> content_flags)
```

Then you can do 

```cpp
if (content_flags.empty()) {
    return true;
}
```

Reply:
The other thing I'm not seeing included in the runtime filtering here is that expansions is not included whatsoever in the filtering logic at runtime. That was being enforced at the query level